app/controllers/foreman_tasks/api/recurring_logics_controller.rb

```diff
@@ -39,9 +39,19 @@ def resource_class
         ForemanTasks::RecurringLogic
       end
 
+      api :POST, '/recurring_logics/bulk_destroy', N_('Delete recurring logics by search query')
+      param :search, String, :desc => N_('Search query'), :required => true
+      def bulk_destroy
+        scope = resource_scope.search_for(params[:search])
```

Comment:
We need to check if `params[:search]` is actually provided. Otherwise it matches *all* recurring logics.

Reply:
`:required => true` doesn't this block it?

Reply:
It just shows up in the api documentation as required, but you have to enforce it yourself. I just accidentally wiped all my recurring logics :)

Reply:
Oh no! Added an `ArgumentError`